Collector/Collector.php

```diff
@@ -18,7 +18,7 @@
  *
  * @internal
  */
-class Collector extends DataCollector
+final class Collector extends DataCollector
```

Comment:
we mock this class in tests, which is failing. do we need to mock this or can we mock a base class instead?

Reply:
Do tests pass if we don't use a mock here?

Reply:
From what i have seen we can easily directly use the collector object without mocking, so would keep this final and remove mock usage in tests

Reply:
i tried that, pushed things, but its rather a lot of things failing right now. don't have more time now to fix things. but the mocking was rather excessive, a bit less isolation and more real tests would probably be good.